app/models/filters/filter_base.rb

```diff
@@ -624,6 +628,19 @@ def available_age_ranges
       }.invert.freeze
     end
 
+    def available_inactivity_days
+      {
+        30 => '30 days',
+        45 => '45 days',
+        60 => '60 days',
+        90 => '90 days',
+        365 => '1 year',
+        365 * 2 => '2 years',
+        365 * 3 => '3 years',
+        365 * 20 => '20 years',
```

Comment:
HUD is picky about considering leap years in some places -- maybe something like `1.year.in_days.round`?

Reply:
Won't that always return 365?

To handle leap years, I think the issue is that we shouldn't do the date subtraction as `days`, we need to do it as `months` or `years`. Testing on 2020 which was a leap year:
```
irb(main):304:0> (Date.parse('01-01-2020') - 12.months)
=> Tue, 01 Jan 2019
irb(main):305:0> (Date.parse('01-01-2021') - 12.months)
=> Wed, 01 Jan 2020
irb(main):306:0> (Date.parse('01-01-2020') - 365.days)
=> Tue, 01 Jan 2019
irb(main):307:0> (Date.parse('01-01-2021') - 365.days)
=> Thu, 02 Jan 2020
irb(main):309:0> (Date.parse('01-01-2020') - 1.year)
=> Tue, 01 Jan 2019
irb(main):308:0> (Date.parse('01-01-2021') - 1.year)
=> Wed, 01 Jan 2020
```

I think I know how to fix, I will leave a comment there..

Reply:
Sorry -- I wasn't clear... Yes, 1 year will always be 365 -- but `20.years.in_days.round != 365 * 20` -- I was just suggesting a pattern

Reply:
Oh I see. Well I think we should really be counting the specific the leap years that happened (or didnt) in the 1 or 2 or 20 year lookback from that `start_date`. https://github.com/greenriver/hmis-warehouse/pull/2186/commits/de38807a653e83a5442b38d809bc3e506d7eee78 should handle that!

Reply:
Isn't de38807 just subtracting days from a date?  I think Sandy's correct, we should be subtracting years (or months) if it's >= 365 days.  Not sure the best way to handle that.  One thing that comes to mind would be a lookup of days as you have it to some method.  Something like `{ 45 => 45.days, 365 => 1.years, 730 => 2.years }` and then use that when subtracting.  Also, I'm not sure how much being off by a day over 4 years is going to matter in this case.

Reply:
> Isn't https://github.com/greenriver/hmis-warehouse/commit/de38807a653e83a5442b38d809bc3e506d7eee78 just subtracting days from a date?

No, it's subtracting  `365.days.in_months.round.months` which returns `12.months` as a duration. Anything over 90 days gets converted to months. I figured we keep 90 and below as days, not months, because that's what we're communicating to the user.